android/src/main/java/com/swmansion/reanimated/layoutReanimation/AnimationsManager.java

```diff
@@ -512,16 +356,150 @@ public void updateLayout(
                 + parentTag
                 + " as a parent, but its Manager doesn't implement IViewManagerWithChildren");
       }
-      if (parentViewManagerWithChildren != null
-          && !parentViewManagerWithChildren.needsCustomLayoutForChildren()) {
+      if (!parentViewManagerWithChildren.needsCustomLayoutForChildren()) {
         viewToUpdate.layout(x, y, x + width, y + height);
       }
     } else {
       viewToUpdate.layout(x, y, x + width, y + height);
     }
   }
 
+  public boolean hasAnimationForTag(int tag, String type) {
+    return mNativeMethodsHolder.hasAnimationForTag(tag, type);
+  }
+
   public boolean isLayoutAnimationEnabled() {
     return mNativeMethodsHolder != null && mNativeMethodsHolder.isLayoutAnimationEnabled();
   }
+
+  private boolean removeOrAnimateExitRecursive(View view, ViewGroup parent, boolean shouldRemove) {
+    int tag = view.getId();
+    boolean hasExitAnimation = hasAnimationForTag(tag, "exiting");
+    boolean wantAnimateExit = hasExitAnimation;
+
+    // we might want to keep this view around
+    // because one of the (children's) children
+    // has an exiting animation
+    if (view instanceof ViewGroup) {
+      ViewGroup viewGroup = (ViewGroup) view;
+      ArrayList<View> children = new ArrayList<>();
```

Comment:
We can just iterate over children in reverse order (from count-1 down to 0) and then copying children's array wouldn't be needed

Reply:
Can we do the same in iOS part to avoid copying?